lab_2/main.py

```diff
@@ -34,7 +55,30 @@ def fill_lcs_matrix(first_sentence_tokens: tuple, second_sentence_tokens: tuple)
     :param second_sentence_tokens: a tuple of tokens
     :return: a lcs matrix
     """
-    pass
+    checks: List[Callable] = [
+        lambda: isinstance(first_sentence_tokens, tuple),
+        lambda: isinstance(second_sentence_tokens, tuple),
+        lambda: first_sentence_tokens,
+        lambda: second_sentence_tokens,
+        lambda: all((isinstance(el, str) for el in x + y))
+    ]
+
+    x, y = first_sentence_tokens, second_sentence_tokens
+
+    for check in checks:
+        if not check():
+            return []
+
+    lcs = create_zero_matrix(len(x), len(y))
+
+    for i, w1 in enumerate(x):
+        for j, w2 in enumerate(y):
+            if w1 == w2:
+                lcs[i][j] = lcs[i - 1][j - 1] + 1
```

Comment:
what happens if i == 0 or j == 0?

Reply:
Well, I got you. 

If i == 0 or j == 0, we start looking at the right end of the matrix, so indices in `lcs[i - 1][j -1] + 1` become negative (both or only of them, it doesn't really matter). This behavior can be handled using one more if condition but...

... 1) We start iterating over the first row and don't mutate other rows so we don't really bother about them even if we get lcs[-1][...], those values are all null at this point of time and become 1 if w1 == w2
2) Similar things happen with j == 0. Iterating over j and looking at it when it equals zero, we can notice that we mutate only the first column of the matrix, i.e. the first elements of each list so why should we care about the last ones? They are still unreachable for us at this moment.

Reply:
Agree